pymarc/marc8.py

```diff
@@ -97,24 +110,20 @@ def translate(self, marc8_string):
                         if pos == len(marc8_string):
                             break
 
-
-            def is_multibyte(charset):
-                return charset == 0x31
-
             mb_flag = is_multibyte(self.g0)
 
             if mb_flag:
-                code_point = (ord(marc8_string[pos:pos+1]) * 65536 +
-                              ord(marc8_string[pos+1:pos+2]) * 256 +
-                              ord(marc8_string[pos+2:pos+3]))
+                code_point = (
+                    (ord(marc8_string[pos:pos+1]) << 16) +  # shift 2 bytes
+                    (ord(marc8_string[pos+1:pos+2]) << 8) +  # shift 1 byte
+                    (ord(marc8_string[pos+2:pos+3]))
+                )
                 pos += 3
             else:
                 code_point = ord(marc8_string[pos:pos+1])
                 pos += 1
 
-            if (code_point < 0x20 or
-                (code_point > 0x80 and code_point < 0xa0)):
-                uni = unichr(code_point)
+            if code_point < 0x20 or 0xa0 > code_point > 0x80:
```

Comment:
This syntax for testing greater-than and less-than at the same time is also new to me. Very cool.